cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -333,7 +333,7 @@ private Datastore createDatastore(String dbName, KeyProvider provider) throws Da
             // if it does not exist
 
             //Pass database directory, database name, and SQLCipher key provider
-            DatastoreImpl ds = new DatastoreImpl(dbDirectory, dbName, provider);
+            QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```

Comment:
But how does the manager know which concrete type to return when `getDatastore()` is invoked? It can't tell in advance whether you want to call Query/Index methods, Attachment, or something else. You have:
```
QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```